### PR TITLE
[INCR 75] only require futures for python 2.7

### PR DESCRIPTION
**Pull request by singuliere**

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @singuliere! I've created [OSPR-3064](https://openedx.atlassian.net/browse/OSPR-3064) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by singuliere** [comment]
The CI errors are unrelated to this PR

```
line 2411, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/toxbat/requirements.py", line 38, in <module>
    from pip.download import PipSession
ImportError: No module named download
The command "tox" exited with 1.
```
Is it a known problem?

**Comment by natabene** [comment]
@singuliere Thank you for your contribution. Can you let me know when tests pass?

**Comment by singuliere** [comment]
blocked by https://github.com/edx/xblock-sdk/pull/146

**Comment by natabene** [comment]
@singuliere Thanks for the update!

**Comment by singuliere** [comment]
rebased on top of #146 to verify it improves the situation

**Comment by singuliere** [comment]
@natabene this PR still contains all commits from #146 (required for tests to pass) and is therefore not yet fit for review. It will be rebased once #146 is merged. Thank you for the kind attention on each and every contribution: it is very reassuring to know you're around to make sure nothing is left unattended!

**Comment by natabene** [comment]
@singuliere Thanks for the update and your kind words, let me know once it is ready, so we can start reviewing.

**Comment by jmbowman** [comment]
#146 has been merged, you should be clear to rebase now.

**Comment by singuliere** [comment]
rebased & repushed :-)

File: requirements/base.in [hunk 1]
```diff
@@ -1,3 +1,4 @@
+-c ../constraints.txt
```
Comment (jmbowman) [inline requirements/base.in]: Could you move these new constraint lines between the file description comments and the requirements?  It's nice to have the summary of what the file is for right at the top, and matches what we've been doing elsewhere.

Comment (singuliere) [reply requirements/base.in]: Sure, done. And also re-ran `make upgrade` to verify I did not accidentally introduced a typo :-)

**Comment by edx-webhook** [comment]
@singuliere 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/xblock-sdk&entry.867055334=https://github.com/edx/xblock-sdk/pull/144&entry.1484655318=https://github.com/singuliere&entry.752974735=2019-02-10+23:15&entry.1917517419=2019-02-22+16:20&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**